### PR TITLE
[NT-735] Sign in with Apple

### PR DESCRIPTION
**Pull request by Scollaco**
# 📲 What

Contains the PRs with the Sign In With Apple work
#1131 - Continue with Apple UI
#1152 - (1/2) Sign in with Apple mutation
#1158  - (2/2) Sign in with Apple
#1163  - Sign in with Apple event
#1168 - Add Loader to Login/Signup Page

# ⏰ TODO

- [x] Add Loading Indicator to the Sign up/Login screen ([JIRA](https://kickstarter.atlassian.net/browse/NT-1136))

👋 
GIF

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: KsApi/MockService.swift [hunk 1]
```diff
@@ -1244,6 +1249,14 @@
       return SignalProducer(value: GraphMutationEmptyResponseEnvelope())
     }
 
+    internal func signInWithApple(input _: SignInWithAppleInput)
+      -> SignalProducer<SignInWithAppleEnvelope, GraphError> {
+      if let error = self.signInWithAppleResult?.error {
+        return SignalProducer(error: error)
+      }
+      return SignalProducer(value: self.signInWithAppleResult?.value ?? SignInWithAppleEnvelope.template)
```
Comment (justinswart) [inline KsApi/MockService.swift:1257]: I don't really think we should ever return defaults from these things 🤷‍♂️ but I know it's been done in a couple places.

File: KsApi/mutations/inputs/SignInWithAppleInput.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+public struct SignInWithAppleInput: GraphMutationInput {
```
Comment (justinswart) [inline KsApi/mutations/inputs/SignInWithAppleInput.swift:3]: If we conform `SignInWithAppleInput` to `Encodable` then we don't need the `toInputDictionary()` function below.

File: Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift [hunk 3]
```diff
@@ -421,14 +487,24 @@ internal final class LoginToutViewController: UIViewController, MFMailComposeVie
 
 extension LoginToutViewController: TabBarControllerScrollable {
   func scrollToTop() {
-    if let scrollView = self.view.subviews.first as? UIScrollView {
```
Comment (justinswart) [inline Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift:424]: How about:
```swift
func scrollToTop() {
  guard let scrollView = self.view.subviews
    .first(where: { view in view is UIScrollView })
    .flatMap({ view in view as? UIScrollView})
  else { return }

  scrollView.scrollToTop()
}
```

File: KsApi/mutations/inputs/SignInWithAppleInput.swift (hunk 2 above)
Comment (ifbarrera) [reply KsApi/mutations/inputs/SignInWithAppleInput.swift:3]: Done! Also noticed we were missing tests for this, so I added those too.

**Pull request merged**